File: src/Core/FinancialEvents/Schemas/FinancialEvents.py
from dataclasses import dataclass
from typing import Sequence

from Core.FinancialEvents.Schemas.Grouping import FinancialGrouping
from Core.FinancialEvents.Schemas.IdentifierRelationship import IdentifierRelationship


@dataclass
class FinancialEvents:
    """Canonical core container: groupings plus identifier relationships.

    Pipeline and ApplyIdentifierRelationshipsService work with this type.
    """

    Groupings: Sequence[FinancialGrouping]
    IdentifierRelationships: Sequence[IdentifierRelationship]

File: lib/core/schemas/FinancialEvents.ts
import type { FinancialGrouping } from "./Grouping.ts";
import type { IdentifierRelationshipAny } from "./IdentifierRelationship.ts";

export type FinancialEvents = {
	groupings: FinancialGrouping[];
	identifierRelationships: IdentifierRelationshipAny[];
};
